libraries/chain/include/eosio/chain/authority.hpp

```diff
@@ -119,31 +139,51 @@ inline bool validate( const Authority& auth ) {
    static_assert( std::is_same<decltype(auth.threshold), uint32_t>::value &&
                   std::is_same<weight_type, uint16_t>::value &&
                   std::is_same<typename decltype(auth.keys)::value_type, key_weight>::value &&
-                  std::is_same<typename decltype(auth.accounts)::value_type, permission_level_weight>::value,
+                  std::is_same<typename decltype(auth.accounts)::value_type, permission_level_weight>::value &&
+                  std::is_same<typename decltype(auth.waits)::value_type, wait_weight>::value,
                   "unexpected type for threshold and/or weight in authority" );
 
-   if( ( auth.keys.size() + auth.accounts.size() ) > (1 << 16) )
+   if( ( auth.keys.size() + auth.accounts.size() + auth.waits.size() ) > (1 << 16) )
       return false; // overflow protection (assumes weight_type is uint16_t and threshold is of type uint32_t)
```

Comment:
The assumptions of this comment could be static_assert'ed

Reply:
The assumptions are static_assert'ed. See the lines directly above it starting at line 139.

Reply:
So it is.